test/pgtest-mz/transactions.pt

```diff
@@ -0,0 +1,55 @@
+# Test implicit and explicit transaction semantics in ways that
+# MZ differs from PG.
+# See pgtest/transactions.pt for more details
+
+# Verify implicit transactions are properly upgraded
+send
+Query {"query": "CREATE TABLE t (a INT)"}
+Parse {"query": "INSERT INTO t VALUES (1)"}
+Bind
+Execute
+Sync
+----
+
+until err_field_typs=M ignore=RowDescription
+ReadyForQuery
+ReadyForQuery
+----
+CommandComplete {"tag":"CREATE TABLE"}
+ReadyForQuery {"status":"I"}
+ParseComplete
+BindComplete
+CommandComplete {"tag":"INSERT 0 1"}
+ReadyForQuery {"status":"I"}
+
+# PG permits commits writes that are part of read-only txns, but
```

Comment:
Wait what? PG allows writes in a read only txn?

Reply:
The behavior here is like:
1. Implicit txn (`READ WRITE`).
2. Upgraded to explicit txn
3. Txn set to `READ ONLY`.

This is equivalent to:

```sql
BEGIN;
INSERT INTO t VALUES (1);
BEGIN READ ONLY;
COMMIT;
```